amr-wind/immersed_boundary/complex_terrain/flat_ops.H

```diff
@@ -72,6 +78,66 @@ struct InitDataOp<Flat>
     }
 };
 
+template <>
+struct ComputeSourceOp<Flat>
+{
+    void operator()(
+        Flat::DataType& data,
+        const int lev,
+        const amrex::MFIter& mfi,
+        const amrex::Geometry& geom)
+    {
+        Field& source = data.sim().repo().get_field("ib_src_term");
+        const Field& level = data.sim().repo().get_field("ib_levelset");
+        const Field& normal = data.sim().repo().get_field("ib_normal");
+        const Field& velocity = data.sim().repo().get_field("velocity");
+
+        const auto& bx = mfi.tilebox();
+        const auto& varr = velocity(lev).array(mfi);
+        const auto& phi_arr = level(lev).array(mfi);
+        const auto& norm_arr = normal(lev).array(mfi);
+        auto sarr = source(lev).array(mfi);
+
+        const amrex::Real cd = data.meta().drag_coefficient;
+        const amrex::Real eps = data.meta().eps_spread;
+
+        amrex::ParallelFor(
+            bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+                // A lot of this is unnecessary for a flat plate, but is
+                // general enough that it could be broken off into its own
+                // function at some point.
+                const auto gauss_fac = actuator::utils::gaussian1d(
+                    amrex::Math::abs(phi_arr(i, j, k)), eps);
+                const amrex::Real u = varr(i, j, k, 0);
+                const amrex::Real v = varr(i, j, k, 1);
+                const amrex::Real w = varr(i, j, k, 2);
+                const amrex::Real vmag = std::sqrt(u * u + v * v + w * w);
```

Comment:
This looks fine.  However if you want to reduce the syntax a lot of these tensor and vector operations are predefined for the `vs::Vector` and vs::Tensor` types through c++operators.  We use them a lot in the actuator code and it makes writing the math less prone to error. [example](https://github.com/Exawind/amr-wind/blob/661a901bb10461ac9722b8956bc6259b0fa46964/amr-wind/wind_energy/actuator/disk/uniform_ct_ops.cpp#L291-L310).